crypto/keygen/keygen.go

```diff
@@ -316,14 +342,22 @@ func GenerateKeyChain(id hotstuff.ID, validFor []string, crypto string, ca *x509
 	certPEM := CertToPEM(cert)
 
 	var privateKey hotstuff.PrivateKey
+	var publicKey hotstuff.PublicKey
 	switch crypto {
 	case "ecdsa":
 		privateKey = ecdsaKey
+		publicKey = privateKey.Public()
 	case "bls12":
 		privateKey, err = bls12.GeneratePrivateKey()
 		if err != nil {
 			return KeyChain{}, fmt.Errorf("failed to generate bls12-381 private key: %w", err)
 		}
+		publicKey = privateKey.Public()
+	case "eddsa":
```

Comment:
Could we replace these case strings with their corresponding `{ecdsa, bls12, eddsa}.PublicKeyFileType`